lib/pos_blockchain.go

```diff
@@ -181,10 +181,14 @@ func (bc *Blockchain) validateBlockGeneral(desoBlock *MsgDeSoBlock) error {
 
 // validateBlockHeight validates the block height for a given block. It checks
 // that this block height is exactly one greater than the current block height.
-// TODO: Are we sure that's the correct validation here?
 func (bc *Blockchain) validateBlockHeight(desoBlock *MsgDeSoBlock) error {
-	// TODO: Implement me
-	return errors.New("IMPLEMENT ME")
+	if desoBlock.Header.Height != bc.GetBestChainTip().Header.Height+1 {
+		return RuleErrorInvalidPoSBlockHeight
+	}
+	// TODO: What if this is a fork? Should this check be more lenient and only check
```

Comment:
Not sure if you do this anywhere else. But you'll also want to verify that the block height is past the PoS cutover fork height somewhere. This feels like a natural place to make that check